### PR TITLE
strip_reflections, strip_modalities tactics

### PR DESCRIPTION
**Pull request by Alizter**
As discussed in #1481

Perhaps it would be better to have a `strip_using` tactic where we provide an induction principle, than all three become special cases of that.

I also wonder why `strip_truncations` is so good at what it does. Replacing it in `Classes/implementations/HProp_lattice.v` for example breaks things. Projective.v also breaks.

### CODE REVIEW

File: theories/Projective.v [hunk 1]
```diff
@@ -154,7 +154,7 @@ Section AC_oo_neg1.
     pose (X := Build_HSet (Tr 0 A)).
     pose proof ((equiv_isoprojective_hasochoice _ X)^-1 (AC X)) as P.
     pose proof (P A _ X _ idmap tr _) as F; clear P.
-    strip_truncations.
+    revert F; refine (Trunc_ind@{_ _} _ _); intros F.
```
Comment (mikeshulman) [inline theories/Projective.v:157]: Why this change?  (Sorry if I missed it.)

**Comment by mikeshulman** [comment]
Would it be worth including `O_indpaths` too?

File: theories/Modalities/Modality.v [hunk 2]
```diff
@@ -208,6 +208,19 @@ Proof.
...
 
+(** A tactic that generalizes [strip_reflections] to modalities. *)
+Ltac strip_modalities :=
+  (** Search for hypotheses of type [O X] for some [O] such that the goal is [O]-local. *)
+  progress repeat
+    match goal with
+    | [ T : _ |- _ ]
+      => revert_opaque T;
+        (** The dependent case requires that [O] be a modality. *)
+        refine (@O_ind@{_ _ _} _ _ _ _ _ _ _);
```
Comment (jdchristensen) [inline theories/Modalities/Modality.v:219]: I wonder if we should try `O_rec` *and* `O_ind` in the modality case?  Maybe there's an advantage to using `O_rec` when it applies?  Also, it would mean that the user can type `strip_modalities` and it would clear reflective subuniverses and modalities, when both are present.

Also:  why the `@{_ _ _}`?

Comment (Alizter) [reply theories/Modalities/Modality.v:219]: I was experimenting with collapsing universes so it could faithfully replace strip_truncations. Couldn't get it to work however.

**Comment by jdchristensen** [comment]
@Alizter Would you like me to make a few changes to finish this off?

**Comment by Alizter** [comment]
@jdchristensen that would be good. I haven't got so much time this week.

File: theories/Projective.v (hunk 1 above)
Comment (jdchristensen) [reply theories/Projective.v:157]: I'm guessing it slipped in by accident.  I'm reversing it.

File: theories/Modalities/Modality.v (hunk 2 above)
Comment (jdchristensen) [reply theories/Modalities/Modality.v:219]: I'm having strip_reflections to O_rec and O_indpaths, and strip_modalities do those two plus O_ind.

**Comment by jdchristensen** [comment]
I think I've handled the remaining issues.  I tested both new tactics, covering all of the cases.  I'll wait for one "LGTM" before merging.

**Comment by jdchristensen** [comment]
I just added a commit that demonstrates strip_reflections by using it in a few places that had the "revert; apply ...; intro" idiom.  (I also removed a few nearby rewrites.)

**Comment by jdchristensen** [comment]
Merging, but if there is feedback later, I'm happy to make fixes.
